File: scripts/utils/platform.py
import platform

WINDOWS = "windows"
LINUX = "linux"
MACOS = "macos"

_system = platform.system().lower()
if _system == "windows":
    SYSTEM = WINDOWS
    EXT = ".exe"
    LIB_EXT = ".dll"
    VCPKG_TRIPLET = "x64-windows-release"
elif _system == "linux":
    SYSTEM = LINUX
    EXT = ""
    LIB_EXT = ".so"
    VCPKG_TRIPLET = "x64-linux-release"
elif _system == "darwin":
    SYSTEM = MACOS
    EXT = ""
    LIB_EXT = ".dylib"
    VCPKG_TRIPLET = "x64-osx-release"
else:
    SYSTEM = "unknown"
    EXT = ".unknown"
    LIB_EXT = ".unknown"
    VCPKG_TRIPLET = "unknown"

X64 = "x64"
ARM64 = "arm64"

_arch = platform.machine().lower()
if _arch in ("x86_64", "amd64"):
    ARCH = X64
elif _arch in ("arm64", "aarch64"):
    ARCH = ARM64
else:
    ARCH = "unknown"

File: src/telebot/utils/platform.cpp
#include "telebot/utils/platform.h"

namespace telebot::utils::platform {

static std::string get_system() {
    #if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
        return WINDOWS;
    #elif __APPLE__
        return MACOS;
    #elif __linux__
        return LINUX;
    #else
        return UNKNOWN;
    #endif
}

static std::string get_arch() {
    #if defined(__x86_64__) || defined(_M_X64)
        return X64;
    #elif defined(__aarch64__) || defined(_M_ARM64)
        return ARM64;
    #endif
}

static std::string get_lib_ext() {
    #if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
        return DLL;
    #elif __APPLE__
        return DYLIB;
    #elif __linux__
        return SO;
    #else
        return UNKNOWN;
    #endif
}

const std::string SYSTEM = get_system();
const std::string ARCH = get_arch();
const std::string LIB_EXT = get_lib_ext();

void init() {

}

} // namespace telebot::utils::platform
